### PR TITLE
Ee 861

### PR DESCRIPTION
**Pull request by MatthewAlexanderBeck**

### CODE REVIEW

File: app/controllers/EmployerContributionController.scala [hunk 1]
```diff
@@ -64,10 +66,11 @@ class EmployerContributionController @Inject()(
 
         value => {
           for {
-          updatedAnswers <- Future.fromTry(request.userAnswers.set(EmployerContributionPage(year, index), value))
-          _              <- sessionRepository.set(updatedAnswers)
-        } yield Redirect(navigator.nextPage(EmployerContributionPage(year, index), mode, updatedAnswers))
-      }
-     )
+            updatedAnswers <- Future.fromTry(request.userAnswers.set(EmployerContributionPage(year, index), value))
+            professionalBodies <- professionalBodiesService.professionalBodies()
```
Comment (garethgooch) [inline app/controllers/EmployerContributionController.scala]: Can we test for failure here?

File: app/controllers/ExpensesEmployerPaidController.scala [hunk 2]
```diff
@@ -70,8 +72,9 @@ class ExpensesEmployerPaidController @Inject()(
         value => {
           for {
             updatedAnswers <- Future.fromTry(request.userAnswers.set(ExpensesEmployerPaidPage(year, index), value))
+            professionalBodies <- professionalBodiesService.professionalBodies()
```
Comment (garethgooch) [inline app/controllers/ExpensesEmployerPaidController.scala]: And here

File: app/models/TaxYearSelection.scala [hunk 3]
```diff
@@ -84,13 +89,69 @@ object TaxYearSelection extends Enumerable.Implicits {
     }
   }
 
+  def filterSelectedTaxYear(taxYearSelection: Seq[TaxYearSelection], claimYear: String): Seq[TaxYearSelection] = {
```
Comment (garethgooch) [inline app/models/TaxYearSelection.scala]: can claimYear be passed in as a string as this is pulled from the params can be Int

File: app/utils/PSubsUtil.scala [hunk 4]
```diff
@@ -55,4 +56,21 @@ object PSubsUtil {
...
+  def duplicatePsubsUserAnswers(
+                     taxYearSelection: Seq[TaxYearSelection],
+                     userAnswers: UserAnswers,
+                     allPsubs: Map[Int, Seq[PSub]],
+                     psubToDuplicte: PSub): UserAnswers = {
+
+    taxYearSelection.foldLeft(userAnswers)(
+      (userAnswers: UserAnswers, taxYearSelection) => {
+
+        val getPsubsForYear: Option[Seq[PSub]] = allPsubs.get(getTaxYear(taxYearSelection))
```
Comment (garethgooch) [inline app/utils/PSubsUtil.scala]: Have we tested the fails when nothing is returned here

File: app/models/TaxYearSelection.scala (hunk 3 above)
Comment (MatthewAlexanderBeck) [reply app/models/TaxYearSelection.scala]: Resolve this for the current ticket and raise a tech debt ticket for the other instances of year

**Pull request merged**